openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -59,6 +60,62 @@ const Carousel = {
             slidesToScroll: a,
             responsive: responsive_settings
         });
+
+        var availabilityStatuses = {
+            'open': {cls: 'cta-btn--available', cta: 'Read'},
+            'borrow_available': {cls: 'cta-btn--available', cta: 'Borrow'},
+            'borrow_unavailable': {cls: 'cta-btn--unavailable', cta: 'Join Waitlist'},
+            'error': {cls: 'cta-btn--missing', cta: 'No eBook'}
+        };
+
+        var addWork = function(work) {
+            var availability = work.availability.status;
+            var ocaid = work.availability.identifier;
+            var cls = availabilityStatuses[availability].cls;
+            var url = (cls == 'cta-btn--available') ?
+                ('/borrow/ia/' + ocaid) : (cls == 'cta-btn--unavailable') ?
+                    ('/books/' + work.availability.openlibrary_edition) : work.key;
+            var cta = availabilityStatuses[availability].cta;
+            var isClickable = availability == 'error' ? 'disabled' : '';
+
+            return '<div class="book carousel__item slick-slide slick-active" ' +
+                '"aria-hidden="false" role="option">' +
+                '<div class="book-cover">' +
+                  '<a href="' + work.key + '" ' + isClickable + '>' +
```

Comment:
Reviewing again to see if I could shrink this file, and I think the intention of `isClickable` is to add a class to the anchor. There's no `class` attribute before it.

Reply:
is clickable evaluates to the literal `disabled` or to empty string ``

Reply:
Ah, right. I scanned it and saw the evaluation against the string "error" thinking that was one of the possible values.